From d167603702055619ec3e66b4694a84e0221591ca Mon Sep 17 00:00:00 2001
From: Greg Hudson <ghudson@mit.edu>
Date: Sun, 4 Oct 2015 11:09:28 -0400
Subject: [PATCH 1/3] Add RUN_TEST make variable

Add a pre.in variable to simplify running C test programs from
Makefile rules.

ticket: 8266

(cherry-picked from commit 45c6e285be8042ffccdab13b5f78cd5ada8e7973)
---
 src/config/pre.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/config/pre.in b/src/config/pre.in
index da8764e..2d58828 100644
--- a/src/config/pre.in
+++ b/src/config/pre.in
@@ -428,6 +428,12 @@ APPUTILS_LIB	= -lapputils
 KRB5_RUN_ENV=@KRB5_RUN_ENV@
 KRB5_RUN_VARS=@KRB5_RUN_VARS@
 
+# Appropriate command prefix for most C test programs: use libraries
+# from the build tree, avoid referencing the installed krb5.conf and
+# message catalog, and use valgrind when asked.
+RUN_TEST=$(RUN_SETUP) KRB5_CONFIG=$(top_srcdir)/config-files/krb5.conf \
+    LC_ALL=C $(VALGRIND)
+
 #
 # variables for --with-tcl=
 TCL_LIBS	= @TCL_LIBS@
-- 
2.9.3

